client/mysqldump.cc

```diff
@@ -6494,61 +6471,49 @@ static int add_slave_statements(void)
   if (opt_comments)
     fprintf(md_result_file,
             "\n--\n-- start slave statement to make a recovery dump)\n--\n\n");
-  if (multi_source)
-    fprintf(md_result_file, "START ALL SLAVES;\n");
-  else
-    fprintf(md_result_file, "START SLAVE;\n");
+  fprintf(md_result_file, "START ALL SLAVES;\n");
   return(0);
 }
 
 static int do_show_slave_status(MYSQL *mysql_con, int have_mariadb_gtid,
                                 int use_gtid, char* set_gtid_pos)
 {
-  MYSQL_RES *UNINIT_VAR(slave);
+  MYSQL_RES *slave;
   MYSQL_ROW row;
   const char *comment_prefix=
     (opt_slave_data == MYSQL_OPT_SLAVE_DATA_COMMENTED_SQL) ? "-- " : "";
   const char *gtid_comment_prefix= (use_gtid ? comment_prefix : "-- ");
   const char *nogtid_comment_prefix= (!use_gtid ? comment_prefix : "-- ");
+  char gtid_pos[MAX_GTID_LENGTH];
 
   if (mysql_query_with_error_report(mysql_con, &slave,
-                                    multi_source ?
-                                    "SHOW ALL SLAVES STATUS" :
-                                    "SHOW SLAVE STATUS"))
+    "SELECT Connection_name, Master_Host, Master_Port,"
+    "Master_Log_File, Exec_Master_Log_Pos FROM information_schema.SLAVE_STATUS")
```

Comment:
🧐 the index-9 column… This should be
```suggestion
    "SELECT Connection_name, Master_Host, Master_Port, Relay_Master_Log_File,"
    "Exec_Master_Log_Pos FROM information_schema.SLAVE_STATUS")
```